sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/parameters.scala

```diff
@@ -177,9 +177,17 @@ object BindParameters extends ParameterizedQueryProcessor with QueryErrorsBase {
   }
 
   private def bind(p: LogicalPlan)(f: PartialFunction[Expression, Expression]): LogicalPlan = {
-    p.resolveExpressionsWithPruning(_.containsPattern(PARAMETER)) (f orElse {
-      case sub: SubqueryExpression => sub.withNewPlan(bind(sub.plan)(f))
-    })
+    p match {
+      case lp: PlanWithUnresolvedIdentifier =>
```

Comment:
Do we need to do it? `resolveExpressionsWithPruning` already recursively transforms all the tree nodes and handles their expressions.